### PR TITLE
[WEBSITE]: Add Curt Hagenlocher to committers list

### PR DESCRIPTION
**Pull request by alamb**
Update https://arrow.apache.org/committers/ to reflect change in committers list

See https://lists.apache.org/thread/nmsqt8wcphysly90l0conmccrqjk3pdo

### CODE REVIEW

File: _data/committers.yml [hunk 1]
```diff
@@ -256,6 +256,10 @@
   role: Committer
   alias: sunchao
   affiliation: Uber
+- name: Curt Hagenlocher
+  role: Committer
+  alias: curth
```
Comment (alamb) [inline _data/committers.yml:261]: This is the apache alias (not the github alias)

**Review by alamb — COMMENTED** [review]
@CurtHagenlocher could you please verify this is the correct information?

File: _data/committers.yml (hunk 1 above)
Comment (CurtHagenlocher) [reply _data/committers.yml:261]: Looks good

**Comment by alamb** [comment]
Thanks @CurtHagenlocher !

**Pull request merged**